File: birthday_chocolate_hacker_rank.rb
def birthday(s, d, m)
    count = 0
    i = 0
    while i < s.length 

        j = i + m
        seg = s[i...j]
        count += 1 if seg.length == m && seg.sum == d

        i += 1
    end
    count

end

p birthday([1, 2, 1, 3, 2], 3, 2)

File: birthday_chocolate_hacker_rank.js
function adder(a, b) {
    return a + b;
}

function birthday(s, d, m) {
    let count = 0;
    let i = 0;
    while (i < s.length - m) {
        let arr = s.splice(i, i + m);
        if (arr.reduce(adder) == d) {
            count += 1
        }
        i++
    }
    return count;
}